jc/parsers/metamf.py

```diff
@@ -0,0 +1,228 @@
+            datalines.pop(-1)
+
+        # extract each archive into its own list of lines.
+        # archives are separated by a blank line
+        this_archive = []
+        for row in datalines:
+            if row == '':
+                archives.append(this_archive)
+                this_archive = []
+                continue
+
+            this_archive.append(row)
+
+        if this_archive:
+            archives.append(this_archive)
+
+        # iterate through list of archives and parse
+        for archive_item in archives:
+
+            manifests = []
+            this_manifest = {}
+            plines = []
+
+            for i, line in enumerate(archive_item):
+                last = archive_item[-1]
+
+                # handle this line separately
+                if (re.match(r'^\s+inflating\s*:\s*META-INF/MANIFEST.MF', line, re.IGNORECASE)):
+                    k, v = line.split(":", maxsplit=1)
+                    v = re.sub(r'\s', '', v)
+                    manifests.append({k: v})
+                    archive_item.pop(i)
+                    continue
+
+                # if line starts with "space"
+                # begin key multiline value pair concatenation
+                if (re.match(r'\s', line)):
+
+                    # expectation is this "if" sets a key once
+                    if (not this_manifest):
+                        # previous line contains a key
+                        k, v = archive_item[i - 1].split(":", maxsplit=1)
+                        v = v + line
+                        v = re.sub(r'\s', '', v)
+                        this_manifest = {k: v}
+                        plines.append(i - 1)
+                        plines.append(i)
+
+                    # continue key multiline value pair concatenation
+                    else:
+                        plines.append(i)
+                        linecmp = line
+                        for k, v in this_manifest.items():
+                            line = v + line
+                            line = re.sub(r'\s', '', line)
+                        this_manifest.update({k:line})
+
+                        if linecmp is not last:
+                            nextline = archive_item[i + 1]
+                            # if next line starts with not "space",
+                            # end key multiline value pair concatenation
+                            if (re.match(r'\S', nextline)):
+                                manifests.append(this_manifest)
+                                this_manifest = False
+                            else:
+                                manifests.append(this_manifest)
+
+            # pop key multiline value pair lines
+            if plines:
+                for p in reversed(plines):
+                    archive_item.pop(p)
+
+            # all other key value pairs
+            for i, line in enumerate(archive_item):
+                k, v = line.split(":", maxsplit=1)
+                v = v.rstrip()
```

Comment:
I think just doing a single `v = v.strip()` will do both left and right strip in a single line.